### PR TITLE
fix: Eth API: make block parameter parsing sounder.

### PR DESCRIPTION
**Pull request by raulk**
## Related Issues

#10426

## Proposed Changes

* Implements the solution as described in #10426.
* Also checks that parameter is not in the future.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [X] Commits have a clear commit message.
- [X] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: itests/eth_transactions_test.go [hunk 1]
```diff
@@ -270,6 +271,76 @@ func TestContractInvocation(t *testing.T) {
...
+func TestGetBlockByNumber(t *testing.T) {
+	blockTime := 100 * time.Millisecond
+	client, _, ens := kit.EnsembleMinimal(t, kit.MockProofs(), kit.ThroughRPC())
+
+	bms := ens.InterconnectAll().BeginMining(blockTime)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	// install contract
```
Comment (arajasek) [inline itests/eth_transactions_test.go:283]: Why are we deploying a contract / creating new EthAccounts here?

**Review by arajasek — APPROVED** [review]
LGTM, modulo the test change.

I see the "returning future tipset" problem has been resolved in a different PR already, but I would recommend [making this general change ](https://github.com/filecoin-project/lotus/issues/10426) (perhaps stripping the EthAPI of direct ChainStore access entirely). But that's a different changeset.

File: itests/eth_transactions_test.go (hunk 1 above)
Comment (raulk) [reply itests/eth_transactions_test.go]: Yeah, that was overkill.

**Comment by arajasek** [comment]
Post-merge LGTM

**Pull request merged**